sandbox-backend/src/main/java/com/exadel/sandbox/team5/service/impl/DiscountServiceImpl.java

```diff
@@ -62,4 +67,39 @@ public DiscountDto update(DiscountDto discount) {
     public void delete(Long id) {
         discountDAO.deleteById(id);
     }
+
+    @Override
+    public Page<DiscountDto> getByCriteria(DiscountSearchCriteria searchCriteria) {
+        String searchText = getWildcard(searchCriteria.getSearchText());
+        List<Discount> result;
+        List<Long> discountId = new ArrayList<>();
+        if (searchCriteria.getTags() == null || searchCriteria.getTags().isEmpty()) {
+            result = discountDAO.getByCriteria(searchText, searchCriteria.getRate());
+        } else {
+            result = discountDAO.getByCriteriaWithTags(searchText,
+                    searchCriteria.getTags(), searchCriteria.getRate());
+        }
+        result.forEach(x -> discountId.add(x.getId()));
```

Comment:
_Just suggestion:_ You can also use lambda:
```
Set<Long> discountIds = result.stream().map(Discount::getId).collect(Collectors.toSet());
```
instead of 
```
List<Long> discountId = new ArrayList<>();
result.forEach(x -> discountId.add(x.getId()));
```

Reply:
done